pinot-core/src/test/java/com/linkedin/pinot/index/writer/FixedByteWidthRowColDataFileWriterTest.java

```diff
@@ -212,9 +212,53 @@ public void testSpecialCharsForStringReaderWriter() throws Exception {
     for (int i = 0; i < rows; i++) {
       String stringInFile = dataFileReader.getString(i, 0);
       Assert.assertEquals(stringInFile, data[i]);
-      Assert.assertEquals(StringUtils.remove(stringInFile, String.valueOf(V1Constants.Str.STRING_PAD_CHAR)),
-          StringUtils.remove(data[i], String.valueOf(V1Constants.Str.STRING_PAD_CHAR)));
+      Assert.assertEquals(StringUtils.remove(stringInFile, String.valueOf(V1Constants.Str.DEFAULT_STRING_PAD_CHAR)),
+          StringUtils.remove(data[i], String.valueOf(V1Constants.Str.DEFAULT_STRING_PAD_CHAR)));
     }
     file.delete();
   }
+
+  @Test
+  public void testSpecialPaddingCharsForStringReaderWriter() throws Exception {
+    for (int iter = 0; iter < 2; iter++) {
+      char paddingChar = (iter == 0) ? '%' :'\0';
+      final byte[] bytes1 = new byte[]{-17, -65, -67, -17, -65, -67, 32, 69, 120, 101, 99, 117, 116, 105, 118, 101};
+      final byte[] bytes2 = new byte[]{-17, -65, -68, 32, 99, 97, 108, 103, 97, 114, 121, 32, 106, 117, 110, 107, 32, 114, 101, 109, 111, 118, 97, 108};
+      File file = new File("test_single_col_writer.dat");
```

Comment:
there is a setting you can have to delete file on exit. Please add that, so that even if the test throws exceptions, the file is deleted. thanks